core/templates/dev/head/pages/collection_editor/collection_editor_navbar_directive.html

```diff
@@ -71,21 +71,21 @@
               (<[getChangeListCount()]>)
             </span>
           </span>
-          <span ng-if="!isPrivate" title="Publish Changes">
+          <span ng-if="!collectionRightsObject.isPrivate()" title="Publish Changes">
             <i class="material-icons md-18 md-dark oppia-save-publish-button-icon"
                alt="Publish Changes">&#xE2C3;</i>
             <span class="oppia-save-publish-button-label">Publish Changes</span>
           </span>
         </span>
 
         <span ng-if="isSaveInProgress()">
-          <span ng-if="isPrivate" class="oppia-save-publish-button-label">Saving</span>
-          <span ng-if="!isPrivate" class="oppia-save-publish-button-label">Publishing</span>
+          <span ng-if="collectionRightsObject.isPrivate()" class="oppia-save-publish-button-label">Saving</span>
+          <span ng-if="!collectionRightsObject.isPrivate()" class="oppia-save-publish-button-label">Publishing</span>
```

Comment:
Maybe this is a good case for having a ``collectionRightsObject.isPublic()`` method; that would help make code a bit more readable.